### PR TITLE
Addressing comments for #1199

### PR DESCRIPTION
**Pull request by AnomalRoil**
Addressing my own code review comments from #1199 
Most notable change: trying to use plain context.Context instead of otel SpanContext when possible in function APIs.

Note: care must be taken not to reuse a context that's coming from a stream in e.g. broadcasting since it could get canceled too early

### CODE REVIEW

File: internal/chain/beacon/chainstore.go [hunk 1]
```diff
@@ -340,14 +340,14 @@ func (c *chainStore) shouldSync(last *common.Beacon, newB likeBeacon) bool {
 // will follow the chain indefinitely. If peers is nil, it uses the peers of
 // the current group.
 func (c *chainStore) RunSync(ctx context.Context, upTo uint64, peers []net.Peer) {
-	_, span := metrics.NewSpan(ctx, "c.RunSync")
+	ctx, span := metrics.NewSpan(ctx, "c.RunSync")
```
Comment (CluEleSsUK) [inline internal/chain/beacon/chainstore.go:343]: hmm the linter complained at me when I did this

File: internal/core/drand_beacon_control.go [hunk 2]
```diff
@@ -295,8 +295,8 @@ func (bp *BeaconProcess) StartFollowChain(ctx context.Context, req *drand.StartS
 	// ctx, cancel := context.WithCancel(context.Background())
```
Comment (CluEleSsUK) [inline internal/core/drand_beacon_control.go:295]: weird that this is commented out

File: internal/core/drand_beacon_control.go [hunk 3]
```diff
@@ -541,25 +544,21 @@ func (bp *BeaconProcess) chainInfoFromPeers(ctx context.Context, privGateway *ne
 // sendProgressCallback returns a function that sends SyncProgress on the
 // passed stream. It also returns a channel that closes when the callback is
 // called with a beacon whose round matches the passed upTo value.
-func (bp *BeaconProcess) sendProgressCallback(
-	ctx context.Context,
-	stream drand.Control_StartFollowChainServer, // not ideal since we also reuse it for the StartCheckChain
-	upTo uint64,
-	info *chain2.Info,
+func (bp *BeaconProcess) sendProgressCallback(ctx context.Context,
```
Comment (CluEleSsUK) [inline internal/core/drand_beacon_control.go:547]: this formatting looks a bit funky - newline per param would be nicer

Comment (AnomalRoil) [reply internal/core/drand_beacon_control.go]: Granted

**Review by CluEleSsUK — APPROVED** [review]

**Pull request merged**